### PR TITLE
Refactor Vehicle

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Fix bug in R8_SPIN and LR_SPIN that would cause vehicles to spin in wrong direction on certain track elements.
Name parameters
Name flag for additional vehicle spinning sprites
Add comments to code to make function easier to understand.

I've fixed a bug in the vehicle update code so a network version up rev required.

### CODE REVIEW

File: src/openrct2/ride/CableLift.cpp [hunk 1]
```diff
@@ -65,8 +65,8 @@ rct_vehicle * cable_lift_segment_create(sint32 rideIndex,
     current->swinging_car_var_0     = 0;
     current->var_4E                 = 0;
     current->restraints_position    = 0;
-    current->var_BA                 = 0;
-    current->var_B6                 = 0;
+    current->spin_sprite                 = 0;
```
Comment (Gymnasiast) [inline src/openrct2/ride/CableLift.cpp]: alignment

**Comment by AaronVanGeffen** [comment]
Nice find, Duncan! 👍 Cheers for refactoring all this, too.

File: src/openrct2/ride/Vehicle.cpp [hunk 2]
```diff
@@ -6597,22 +6597,22 @@ static void vehicle_update_track_motion_up_stop_check(rct_vehicle * vehicle)
  * This function must be called when the car is running through a non-stopping
  * state block-brake (precondition), which means that the block brake is acting
  * merely as a velocity regulator, in a closed state. When the brake is open, it
- * boosts the train to the speed limit
+ * boosts the train to the speedModifier limit
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp]: revert the comment change woops!

**Comment by duncanspumpkin** [comment]
Vehicle.cpp#L3985 update to use track motion flags. #L3705 as well and #L3325, #L2061.

File: src/openrct2/ride/Vehicle.cpp [hunk 3]
```diff
@@ -9457,6 +9480,118 @@ static sint32 vehicle_update_track_motion_mini_golf(rct_vehicle * vehicle, sint3
...
+    if (vehicleEntry->flags & VEHICLE_ENTRY_FLAG_POWERED_RIDE_UNRESTRICTED_GRAVITY)
+    {
+        if (vehicle->velocity > (vehicle->speed * 0x4000))
+        {
+            // Same code as none powered rides
+            if (*acceleration <= 0)
+            {
+                if (*acceleration >= -500)
+                {
+                    if (vehicle->velocity <= 0x8000)
```
Comment (AaronVanGeffen) [inline src/openrct2/ride/Vehicle.cpp]: Could we collapse these nested conditionals into one block?

Comment (duncanspumpkin) [reply src/openrct2/ride/Vehicle.cpp]: Probably. I was planning on breaking this code out into a separate function as it's used in 3+ places

File: src/openrct2/ride/Vehicle.cpp [hunk 4]
```diff
@@ -7296,80 +7326,80 @@ static void sub_6D63D4(rct_vehicle * vehicle)
...
-        if (vehicle->var_C5 != al)
+        if (vehicle->animation_frame != al)
         {
-            vehicle->var_C5 = al;
+            vehicle->animation_frame = al;
             vehicle_invalidate(vehicle);
         }
         break;
-    case 3: // loc_6D6482
+    case VEHICLE_ENTRY_ADDITIONAL_ANIMATION_3: // loc_6D6482
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp]: type 5 is helicar, animation type 3 is rowboats
type 8 is monorail cycles, type 6 is the observation tower
type 9 is 4D coaster, and I don't know about 2, 4, or 7

**Comment by duncanspumpkin** [comment]
Yes i was trying to name the flags yesterday for swinging. They also effect how much it swings as well as how many sprites (i guess that's expected since it wouldn't make sense otherwise). I think i might end up just naming them swinging 1-3 with a note stating what they change

File: src/openrct2/ride/Ride.cpp [hunk 5]
```diff
@@ -4557,11 +4557,11 @@ static rct_vehicle *vehicle_create_car(
         sprite_move(chosenLoc.x, chosenLoc.y, z, (rct_sprite*)vehicle);
     } else {
         regs.dl = 0;
-        if (vehicleEntry->flags & VEHICLE_ENTRY_FLAG_28) {
+        if (vehicleEntry->flags & VEHICLE_ENTRY_FLAG_CHAIR_LIFT) {
```
Comment (Gymnasiast) [inline src/openrct2/ride/Ride.cpp]: We write 'chairlift' in most (all?) other places.

**Pull request merged**